api/buckets/client/client_test.go

```diff
@@ -30,23 +29,131 @@ func TestClient_Init(t *testing.T) {
 	ctx, client, done := setup(t)
 	defer done()
 
-	buck, err := client.Init(ctx, "mybuck")
+	buck, err := client.Init(ctx, c.WithName("mybuck"))
 	require.Nil(t, err)
 	assert.NotEmpty(t, buck.Root)
 	assert.NotEmpty(t, buck.Root.Key)
+	assert.NotEmpty(t, buck.Root.Name)
 	assert.NotEmpty(t, buck.Root.Path)
 	assert.NotEmpty(t, buck.Root.CreatedAt)
 	assert.NotEmpty(t, buck.Root.UpdatedAt)
 	assert.NotEmpty(t, buck.Links)
 	assert.NotEmpty(t, buck.Seed)
+
+	pbuck, err := client.Init(ctx, c.WithName("myprivatebuck"), c.WithPrivate(true))
+	require.Nil(t, err)
+	assert.NotEmpty(t, pbuck.Root)
+	assert.NotEmpty(t, pbuck.Root.Key)
+	assert.NotEmpty(t, pbuck.Root.Name)
+	assert.NotEmpty(t, pbuck.Root.Path)
+	assert.NotEmpty(t, pbuck.Root.CreatedAt)
+	assert.NotEmpty(t, pbuck.Root.UpdatedAt)
+	assert.NotEmpty(t, pbuck.Links)
+	assert.NotEmpty(t, pbuck.Seed)
+}
+
+func TestClient_InitWithCid(t *testing.T) {
+	t.Parallel()
+	ctx, client, done := setup(t)
+	_ = client
+	defer done()
+
+	file1, err := os.Open("testdata/file1.jpg")
+	require.Nil(t, err)
+	defer file1.Close()
+	file2, err := os.Open("testdata/file2.jpg")
+	require.Nil(t, err)
+	defer file2.Close()
+
+	ipfs, err := httpapi.NewApi(util.MustParseAddr("/ip4/127.0.0.1/tcp/5001"))
+	require.NoError(t, err)
+	p, err := ipfs.Unixfs().Add(
+		ctx,
+		ipfsfiles.NewMapDirectory(map[string]ipfsfiles.Node{
+			"file1.jpg": ipfsfiles.NewReaderFile(file1),
+			"folder1": ipfsfiles.NewMapDirectory(map[string]ipfsfiles.Node{
+				"file2.jpg": ipfsfiles.NewReaderFile(file2),
+			}),
+		}),
+	)
+	require.NoError(t, err)
+	initCid := p.Cid()
+	buck, err := client.Init(ctx, c.WithCid(initCid))
+	require.NoError(t, err)
+
+	// Assert top level bucket.
+	rep, err := client.ListPath(ctx, buck.Root.Key, "")
+	require.Nil(t, err)
+	assert.True(t, rep.Item.IsDir)
+	topLevelNames := make([]string, 3)
+	for i, n := range rep.Item.Items {
+		topLevelNames[i] = n.Name
+	}
+	assert.Contains(t, topLevelNames, "file1.jpg")
+	assert.Contains(t, topLevelNames, "folder1")
+
+	// Assert inner directory.
+	rep, err = client.ListPath(ctx, buck.Root.Key, "folder1")
+	require.Nil(t, err)
+	assert.True(t, rep.Item.IsDir)
+	assert.Equal(t, 1, len(rep.Item.Items))
+	assert.Equal(t, rep.Item.Items[0].Name, "file2.jpg")
+}
+
+func TestClient_InitWithCid_Private(t *testing.T) {
```

Comment:
Each (relevant) method has a `_Private` flavor that uses bucket encryption

Reply:
Good call, was racing last week :) I split them all out into wrapped test methods.